packages/angular/cli/src/commands/generate/cli.ts

```diff
@@ -162,4 +160,45 @@ export class GenerateCommandModule
 
     return `${commandName}${positionalArgs ? ' ' + positionalArgs : ''}`;
   }
+
+  private async getSchematicsToRegister(): Promise<
+    [schematicName: string, collectionName: string][]
+  > {
+    const [, schematicNameFromArgs] = this.parseSchematicInfo(
+      // positional = [generate, component] or [generate]
+      this.context.args.positional[1],
+    );
+
+    /**
+     * - *key*: schematic name
+     * - *value*: collection name
+     */
+    const schematicsToRegister = new Map<string, string>();
+    const collectionNames = await this.getCollectionNames();
+
+    for (const collectionName of collectionNames) {
+      const workflow = this.getOrCreateWorkflowForBuilder(collectionName);
+      const collection = workflow.engine.createCollection(collectionName);
+      const schematicsInCollection = collection.description.schematics;
+
+      // We cannot use `collection.listSchematicNames()` as this doesn't return hidden schematics.
```

Comment:
Can we add a new `collection.listHiddenSchematicNames()` or `collection.listAllSchematicNames()`?

Reply:
Or maybe 

```ts
listSchematicNames(collection: FileSystemCollectionDesc, includeHidden: boolean): string[];
```

Reply:
https://github.com/angular/angular-cli/pull/22860/commits/ff692ca72acba38015201b2a32b01580ab7a246c